### ISSUE TITLE
Using ruby 2.4 fails

### ISSUE THREAD

**Issue by pcbeard** [wconrad/ftpd#45]
I tried installing the most modern ruby available in Homebrew (ruby 2.4), and kept running into a problem compiling json (1.8.3):

```
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /usr/local/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
/usr/local/opt/ruby/bin/ruby -r ./siteconf20170624-88439-qnl4kq.rb extconf.rb
creating Makefile

current directory: /usr/local/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR=" clean

current directory: /usr/local/lib/ruby/gems/2.4.0/gems/json-1.8.3/ext/json/ext/generator
make "DESTDIR="
compiling generator.c
generator.c:861:25: error: use of undeclared identifier 'rb_cFixnum'
    } else if (klass == rb_cFixnum) {

```

Easily fixed by editing Gemfile.lock to use the latest json (2.1.0):

```
diff --git a/Gemfile.lock b/Gemfile.lock
index d1313a4..81b7532 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -42,7 +42,7 @@ GEM
       rake
       rdoc
       semver
-    json (1.8.3)
+    json (2.1.0)
     jwt (1.5.6)
     memoizer (1.0.1)
     mini_portile2 (2.1.0)
```

**Comment by wconrad** [comment]
I got rid of the `jeweler` gem as a development dependency.  The `jeweler` gem is what was pulling in the `json` gem.  That should take care of the problem you were having running `bundle` against this library in Ruby 2.4.  I'll close this issue, then, but I can't test on homebrew so please let me know if I've made a mistake.

Please note that the tests currently do not pass in Ruby 2.3, because the `double-bag-ftps` gem that ftpd's tests use is currently broken for Ruby 2.4 (see https://github.com/bnix/double-bag-ftps/issues/17).  You'll need to use ruby-2.3 to develop ftpd for the mean time.  I believe ftpd works in ruby-2.4; I just don't have running tests to prove it.

**Comment by wconrad** [comment]
The tests now pass in Ruby 2.4.  Well, sort of pass.  I marked those tests that are broken as "pending"; that doesn't make me very nervous because those few tests do run (and pass) in Rubies 1.9.3 through 2.3.4, and it's pretty unlikely that there'll be a Ruby 2.4 specific bug in that part of the code.

Eventually I may be able to get the `double-bag-ftps` gem fully operational in Ruby 2.4, and then the tests I marked "pending" for Ruby 2.4 can be put back into full service.
